src/Sarif/Visitors/SortingVisitor.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class SortingVisitor : SarifRewritingVisitor
+    {
+        private readonly IDictionary<int, int> ruleIndexMap;
+        private readonly IDictionary<int, int> artifactIndexMap;
+
+        public SortingVisitor()
+        {
+            ruleIndexMap = new ConcurrentDictionary<int, int>();
```

Comment:
>ConcurrentDictionary [](http://example.com/codeflow?start=31&length=20)

If you are proposing to make this thread-safe, write a test to prove it.

Reply:
Visitor traverses entire Sarif log by properties in certain order. I could not think multi threads scenario for this. If its true I will change it to normal dictionary.

Reply:
changed to Dictioanry<>